modules/apps/message-boards/message-boards-web/src/main/java/com/liferay/message/boards/web/internal/display/context/MBEntriesManagementToolbarDisplayContext.java

```diff
@@ -126,6 +130,39 @@ public MBEntriesManagementToolbarDisplayContext(
 		).build();
 	}
 
+	public Map<String, Object> getAdditionalProps() {
+		return HashMapBuilder.<String, Object>put(
+			"deleteEntriesCmd",
+			() -> {
+				if (_trashHelper.isTrashEnabled(
+						_themeDisplay.getScopeGroupId())) {
+
+					return Constants.MOVE_TO_TRASH;
+				}
+
+				return Constants.DELETE;
+			}
+		).put(
+			"editEntryURL",
+			() -> {
+				PortletURL editEntryURL =
+					_liferayPortletResponse.createActionURL();
+
+				editEntryURL.setParameter(
+					ActionRequest.ACTION_NAME, "/message_boards/edit_entry");
+
+				return editEntryURL.toString();
+			}
+		).put(
+			"lockCmd", Constants.LOCK
+		).put(
+			"trashEnabled",
+			() -> _trashHelper.isTrashEnabled(_themeDisplay.getScopeGroupId())
+		).put(
+			"unlockCmd", Constants.UNLOCK
+		).build();
+	}
+
```

Comment:
I see your point but I thought (at least in Lima) that we prefered to have clean JSP's and all the logic in the DisplayContexts. What do you think about it @4lejandrito?

Reply:
I don't think we follow a strict rule. I personally apply the following reasoning depending on the complexity of the Java logic needed to calculate the props values:

* **Simple logic:** I prefer to create them directly in the JSP. It makes it easier to read and I don't have to navigate to another file to know which props are being passed.
* **Complex logic:** This is where I think using the display context comes handy. It allows us to extract the logic into private reusable methods and take advantage of the better source formatting for Java files. It is also generally easier to maintain a Java file than a JSP one.

In this specific case... since we are using a private variable (`_trashHelper`) and a conditional I would have used the display context too.

I would say it's a matter of what's easier and more maintainable in each case.